File: data/multilang_comparison/program_cpp_vector/src/func23.cpp
#include "program_cpp_vector.hpp" 
Array* func23(Array_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   Array* array45;
   if (pCounter > 0) {
      array45 = vars->data.at(--pCounter);
      array45->refC++;
      DEBUG_COPY(array45->id);
   } else {
      array45 = new Array();
      array45->size = 555;
      array45->refC = 1;
      array45->id = 45;
      array45->data = std::vector<unsigned int>(array45->size, 0);
      DEBUG_NEW(array45->id);
   }
   Array_param params0;
   params0.size = 1;
   std::vector<Array*> data_params0(1);
   params0.data = data_params0;
   params0.data.at(0) = array45;
   Array* array46 = func28(&params0, get_path(), loopsFactor);
   DEBUG_RETURN(array46->id);
   if(params0.size > 0) {
   	 params0.data.clear();
   }
   unsigned int loop15 = 0;
   unsigned int loopLimit15 = (50)/4 + 1;
   for(; loop15 < loopLimit15; loop15++) {
      for (int i = 0; i < array45->size; i++) {
         array45->data.at(i)++;
      }
   }
   for (int i = 0; i < array45->size; i++) {
       if (array45->data.at(i) == 41) {
           return array45;
       }
   }
   array46->refC--;
   if(array46->refC == 0) {
      array46->data.clear();
      delete array46;
      DEBUG_FREE(array46->id);
   }
   return array45;
}

File: data/multilang_comparison/program_go/src/func23.go
package main
import "math/rand"
var _ = rand.Intn
func Func23(vars *ArrayParam, loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array45 *Array
   if pCounter > 0 {
       array45 = vars.Data[pCounter-1]
       pCounter--
       array45.RefC++
   } else {
       array45 = &Array{
           Size: 555,
           RefC: 1,
           Id: 45,
           Data: make([]uint32, 555),
       }
   }
   var params0 ArrayParam
   params0.Size = 1
   params0.Data = make([]*Array, params0.Size)
   params0.Data[0] = array45
   var array46 *Array = Func28(&params0, getPath(), loopsFactor);
   params0.Data = nil
   params0.Size = 0
   var loop15 int = 0;
   var loopLimit15 int = (50)/4 + 1;
   for ; loop15 < loopLimit15; loop15++ {
      for i := 0; i < array45.Size; i++ {
          array45.Data[i]++
      }
   }
   for i := 0; i < array45.Size; i++ {
       if array45.Data[i] == 41 {
           return array45
       }
   }
   array46.RefC--
   if array46.RefC == 0 {
       array46.Data = nil
       array46 = nil
   }
   return array45;
}

